Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstandingCustomMesh.cs

```diff
@@ -43,89 +48,230 @@ public bool DrawProcessedMesh
             }
         }
 
-        protected override Material RenderMaterial { get { return MeshMaterial; } }
-        
         /// <summary>
-        /// World anchor used by the custom mesh
+        /// Indicates if the previous import is still being processed.
         /// </summary>
-        public WorldAnchor LocalWorldAnchor { get; private set; }
+        public bool IsImportActive { get; private set; }
 
-        // Privates
-        private bool drawProcessedMesh = true;
-        private bool isImportActive = false;
+        /// <summary>
+        /// The material to use if rendering.
+        /// </summary>
+        protected override Material RenderMaterial { get { return MeshMaterial; } }
+
+        /// <summary>
+        /// To prevent us from importing too often, we keep track of the last import.
+        /// </summary>
         private DateTime timeLastImportedMesh = DateTime.Now;
-        private Mesh customMesh;
 
-        // Functions
+        /// <summary>
+        /// For a cached SpatialUnderstanding.Instance.
+        /// </summary>
+        private SpatialUnderstanding spatialUnderstanding;
+
+        /// <summary>
+        /// The spatial understanding mesh will be split into pieces so that we don't have to 
+        /// render the whole mesh, rather just the parts that are visible to the user.
+        /// </summary>
+        private Dictionary<Vector3, MeshData> meshSectors = new Dictionary<Vector3, MeshData>();
+
+        /// <summary>
+        /// A data structure to manage collecting triangles as we 
+        /// subdivide the spatial understanding mesh into smaller sub meshes.
+        /// </summary>
+        private class MeshData
+        {
+            /// <summary>
+            /// Lists of verts/triangles that describe the mesh geometry.
+            /// </summary>
+            private List<Vector3> Verts = new List<Vector3>();
+            private List<int> Tris = new List<int>();
+
+            /// <summary>
+            /// The mesh object based on the triangles passed in.
+            /// </summary>
+            public Mesh MeshObject;
```

Comment:
If it were a monobehavior to avoid the editor looking at it maybe...  otherwise I'm not sure if it matters.